sdk/mysql/Azure.ResourceManager.MySql/src/autorest.md

```diff
@@ -161,27 +159,6 @@ directive:
   - rename-operation:
       from: Servers_Upgrade
       to: MySqlServers_Upgrade
-  - from: mysql.json
-    where: $.definitions
-    transform: >
-      $.ConfigurationListContent = {
-          "properties": {
-            "value": {
-              "type": "array",
-              "items": {
-                "$ref": "#/definitions/Configuration"
-              },
-              "description": "The list of server configurations."
-            }
-          },
-          "description": "A list of server configurations."
-        };
-      $.ConfigurationListResult.properties.value.readOnly = true;
-    reason: The generator will not treat the model as the schema for a list method without value being a IReadOnlyList. Need to have separate models for input and output.
-  - from: mysql.json
-    where: $.paths['/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/updateConfigurations'].post.parameters[?(@.name === 'value')]
-    transform: >
-      $.schema['$ref'] = $.schema['$ref'].replace('ConfigurationListResult', 'ConfigurationListContent');
```

Comment:
OK removal of these configurations is changing the input parameter from XXXContent to a XXXListResult, which is very weird

Reply:
Previously, the ConfigurationListResult with a settable IList caused GetAll method not recognized. The ConfigurationListContent is a workaround to make it work, but it has a drawback that the input and output of UpdateConfigurations/GetAll method are different, users  need to convert the models when they get a ConfigurationListResult and want to change it a bit, then send it back to update configurations. Therefore, I reverted the change after fix in generator. I will update the model since it can be input as well.

A question to you is that currently UpdateConfigurations (/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.DBforMySQL/servers/{serverName}/updateConfigurations) is put under MySqlServerResource, I think it makes more sense to put it under MySqlConfigurationCollection. Is there a way to do it through autorest.md configurations?